src/components/menu/OutsideClick.tsx

```diff
@@ -0,0 +1,62 @@
+// (C) 2007-2018 GoodData Corporation
+import * as React from 'react';
+
+export interface IOutsideClickHandlerProps {
+    onOutsideClick: (e: MouseEvent) => void;
+    useCapture?: boolean;
+}
+
+export default class OutsideClickHandler extends React.Component<IOutsideClickHandlerProps> {
+    public static defaultProps = {
+        // Set to true by default so that a `stopPropagation` in the
+        // children will not prevent all outside click handlers from firing
+        useCapture: true
+    };
+
+    private wrapperEl: HTMLElement = null;
+
+    public componentWillReceiveProps(prevProps: IOutsideClickHandlerProps) {
+        if (prevProps.onOutsideClick !== this.props.onOutsideClick || prevProps.useCapture !== this.props.useCapture) {
+            this.removeListeners();
+            this.addListeners();
+        }
+    }
+
+    public componentWillMount() {
+        this.addListeners();
+    }
+
+    public componentWillUnmount() {
+        this.removeListeners();
+    }
+
+    public render() {
+        return (
+            <div
+                ref={(el: HTMLElement) => {
```

Comment:
Could you use the new createRef API? See: https://reactjs.org/blog/2018/03/29/react-v-16-3.html
(for all refs in this PR)

Reply:
I used the newer api at the beginning and while there is new enough React version, the typings package is outdated and it does not know about `React.createRef`. I spend few minutes on updating the typings package but it brought lots of typing errors so I think this, along with updating typings & react version, should be handled in separate ticket / PR.